src/software/ai/hl/stp/play/free_kick/free_kick_play_fsm.cpp

```diff
@@ -0,0 +1,317 @@
+{
+    LOG(INFO) << "Time to look for pass expired. Chipping ball.";
+    PriorityTacticVector tactics_to_run = {{}};
+
+    Point ball_pos = event.common.world_ptr->ball().position();
+    std::optional<Robot> robot_kicking_opt =
+        event.common.world_ptr->friendlyTeam().getNearestRobot(ball_pos);
+    if (!robot_kicking_opt.has_value())
+    {
+        LOG(WARNING) << "No robot found to chip the ball during free kick.";
+        return;
+    }
+
+    // Chip towards the friendly farthest up the enemy half,
+    // or the center of the field if no friendly is in the enemy half.
+    Point chip_target(0, 0);
+    for (const Robot &friendly : event.common.world_ptr->friendlyTeam().getAllRobots())
+    {
+        // Skip over the robot kicking the ball
+        if (friendly.id() == robot_kicking_opt->id())
+        {
+            continue;
+        }
+
+        if (friendly.position().x() > chip_target.x())
+        {
+            chip_target = friendly.position();
+        }
+    }
+
+    chip_tactic->updateControlParams(event.common.world_ptr->ball().position(),
+                                     chip_target);
+    tactics_to_run[0].emplace_back(chip_tactic);
+
+    event.common.set_tactics(tactics_to_run);
+}
+
+void FreeKickPlayFSM::lookForPass(const FreeKickPlayFSM::Update &event)
+{
+    PriorityTacticVector tactics_to_run = {{}};
+
+    // Keep the kicker aligned to the ball
+    updateAlignToBallTactic(event.common.world_ptr);
+    tactics_to_run[0].emplace_back(align_to_ball_tactic);
+
+    // Already assigned one tactic to align to ball.
+    // Assign up to 2 receivers and the remaining tactics are defenders
+    setReceiverAndDefenderTactics(tactics_to_run, event, 2, 1);
+
+    // Find the current best pass
+    // Avoid passes to the goalie and the passing robot
+    std::vector<RobotId> robots_to_ignore = {};
+    auto friendly_goalie_id_opt = event.common.world_ptr->friendlyTeam().getGoalieId();
+    if (friendly_goalie_id_opt.has_value())
+    {
+        robots_to_ignore.push_back(friendly_goalie_id_opt.value());
+    }
+    auto robot_with_ball_opt = event.common.world_ptr->friendlyTeam().getNearestRobot(
+        event.common.world_ptr->ball().position());
+    if (robot_with_ball_opt.has_value())
+    {
+        robots_to_ignore.push_back(robot_with_ball_opt.value().id());
+    }
+    best_pass_and_score_so_far =
+        pass_generator.getBestPass(*event.common.world_ptr, robots_to_ignore);
+
+    event.common.set_tactics(tactics_to_run);
+}
+
+bool FreeKickPlayFSM::passFound(const Update &event)
+{
+    double time_since_pass_optimization_start_seconds =
+        (event.common.world_ptr->getMostRecentTimestamp() - pass_optimization_start_time)
+            .toSeconds();
+
+    double abs_min_pass_score =
+        ai_config.shoot_or_pass_play_config().abs_min_pass_score();
+    double min_perfect_pass_score =
+        ai_config.shoot_or_pass_play_config().min_perfect_pass_score();
```

Comment:
What is the plan here for when #3246 is merged in? I believe ShootOrPassPlayConfig is going to be removed and replaced with ShotConfig, which is very different.

Reply:
Both PRs are being aimed to be merged with master, so which ever one is merged second will need to fix it. I can't update this PR to use `ShotConfig`, since it doesn't exist on my branch.